### PR TITLE
Less stringent dependency check for building/installing skimage

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

**Comment by jni** [comment]
@stefanv I don't understand why but I don't think I'm getting the desired output:

```
=========================================================
Warning: Unsatisfied dependencies for using scikit-image!
=========================================================
You need                 You have
---------------------------------------------------------
Cython 0.21              --
=========================================================
Install runtime dependencies using:

  pip install -r requirements.txt
=========================================================

=========================================================
Warning: Unsatisfied dependencies for using scikit-image!
=========================================================
You need                 You have
---------------------------------------------------------
matplotlib 1.1.0         --
=========================================================
Install runtime dependencies using:

  pip install -r requirements.txt
=========================================================

=========================================================
Warning: Unsatisfied dependencies for using scikit-image!
=========================================================
You need                 You have
---------------------------------------------------------
networkx 1.8             --
=========================================================
Install runtime dependencies using:

  pip install -r requirements.txt
```

(and so on in this fashion until everyone is eaten) #futurama

File: setup.py [hunk 1]
```diff
@@ -86,41 +86,52 @@ def get_package_version(package):
...
-                dep_error = ('You need `%s` version %s or later,'
-                             'found version %s.'
-                             % (package_name, min_version,
-                                package_version))
-        if dep_error:
-            raise ImportError(dep_error)
+            if not package_version or LooseVersion(min_version) > LooseVersion(package_version):
+                dep_errors.append((package_name, min_version, package_version))
+
+        if dep_errors:
```
Comment (JDWarner) [inline setup.py]: @jni I think that would be fixed if this block were moved down one level.

Comment (stefanv) [inline setup.py]: Yup, that's it.

File: setup.py [hunk 2]
```diff
@@ -86,41 +86,52 @@ def get_package_version(package):
...
+            print('=========================================================')
+            for (pkg, version_required, version_installed) in dep_errors:
+                print('You need                ', 'You have'                 )
+                print('---------------------------------------------------------')
+                print(('%s %s' % (pkg, version_required)).ljust(24),
+                      version_installed)
+            print('=========================================================')
+            print('Install runtime dependencies using:')
+            print('')
+            print('  pip install -r requirements.txt')
```
Comment (blink1073) [inline setup.py]: I'm not so sure about this.  Wheels are only available for numpy and scipy on OSX.

Comment (blink1073) [inline setup.py]: Perhaps something like "See requirements.txt for a list of dependencies"?

**Comment by stefanv** [comment]
Thanks, hopefully this is fixed now

**Comment by jni** [comment]
@stefanv I don't think it's quite there yet:

You need:

```
=========================================================
Warning: Unsatisfied dependencies for using scikit-image!
=========================================================
You need                 You have
---------------------------------------------------------
networkx 1.8             --
PIL.Image 1.1.7          --
Cython 0.21              --
six 1.3                  --
matplotlib 1.1.0         --
=========================================================
Install runtime dependencies using:

  pip install -r requirements.txt

See DEPENDS.txt for more information.
=========================================================
```

You have:
```
=========================================================
Warning: Unsatisfied dependencies for using scikit-image!
=========================================================
You need                 You have
---------------------------------------------------------
networkx 1.8             --
You need                 You have
---------------------------------------------------------
PIL.Image 1.1.7          --
You need                 You have
---------------------------------------------------------
Cython 0.21              --
You need                 You have
---------------------------------------------------------
six 1.3                  --
You need                 You have
---------------------------------------------------------
matplotlib 1.1.0         --
=========================================================
Install runtime dependencies using:

  pip install -r requirements.txt

See DEPENDS.txt for more information.
=========================================================
```

=P =P =P

**Comment by stefanv** [comment]
Annoyingly incompetent.  But apart from that, let's try a different approach (see https://gitter.im/scikit-image/scikit-image?at=54ff4d0fbc717c2302df96f9)

**Comment by jni** [comment]
@stefanv OMG quotable Gitter!!!! !!! =D =D =D

**Pull request closed without merging**